hedera-node/src/test/java/com/hedera/services/records/TransactionRecordServiceTest.java

```diff
@@ -64,6 +75,70 @@ void setUp() {
 		subject = new TransactionRecordService(txnCtx);
 	}
 
+
+	@Test
+	void externalisesEvmCreateTransactionWithSuccess() {
+		given(result.isSuccessful()).willReturn(true);
+		subject.externaliseEvmCreateTransaction(result);
+
+		verify(txnCtx).setStatus(SUCCESS);
+		verify(txnCtx).setCreateResult(any());
+		verify(txnCtx).addNonThresholdFeeChargedToPayer(anyLong());
+	}
+
+	@Test
+	void externalisesEvmCreateTransactionWithSelfDestruct() {
+		given(result.isSuccessful()).willReturn(false);
+		given(result.getHaltReason()).willReturn(Optional.of(SELF_DESTRUCT_TO_SELF));
+		subject.externaliseEvmCreateTransaction(result);
+
+		verify(txnCtx).setStatus(OBTAINER_SAME_CONTRACT_ID);
+		verify(txnCtx).setCreateResult(any());
+		verify(txnCtx).addNonThresholdFeeChargedToPayer(anyLong());
+	}
+
+	@Test
+	void externalisesEvmCreateTransactionWithInvalidSolidityAddress() {
+		given(result.isSuccessful()).willReturn(false);
+		given(result.getHaltReason()).willReturn(Optional.of(INVALID_SOLIDITY_ADDRESS));
+		subject.externaliseEvmCreateTransaction(result);
+
+		verify(txnCtx).setStatus(ResponseCodeEnum.INVALID_SOLIDITY_ADDRESS);
+		verify(txnCtx).setCreateResult(any());
+		verify(txnCtx).addNonThresholdFeeChargedToPayer(anyLong());
+	}
+
+	@Test
+	void externalisesEvmCreateTransactionWithInvalidSignature() {
+		given(result.isSuccessful()).willReturn(false);
+		given(result.getHaltReason()).willReturn(Optional.of(INVALID_SIGNATURE));
+		subject.externaliseEvmCreateTransaction(result);
+
+		verify(txnCtx).setStatus(ResponseCodeEnum.INVALID_SIGNATURE);
+		verify(txnCtx).setCreateResult(any());
+		verify(txnCtx).addNonThresholdFeeChargedToPayer(anyLong());
+	}
+
+	@Test
+	void externalisesEvmCreateTransactionWithContractRevert() {
+		given(result.isSuccessful()).willReturn(false);
+		subject.externaliseEvmCreateTransaction(result);
+
+		verify(txnCtx).setStatus(CONTRACT_REVERT_EXECUTED);
+		verify(txnCtx).setCreateResult(any());
+		verify(txnCtx).addNonThresholdFeeChargedToPayer(anyLong());
+	}
+
+	@Test
+	void externalisesEvmCallTransactionWithSuccess() {
+		given(result.isSuccessful()).willReturn(true);
+		subject.externaliseEvmCallTransaction(result);
+
+		verify(txnCtx).setStatus(SUCCESS);
+		verify(txnCtx).setCallResult(any());
+		verify(txnCtx).addNonThresholdFeeChargedToPayer(anyLong());
+	}
+
```

Comment:
Instead of passing `any()` or `anyLong()`, you can actually pass the real values